x-pack/platform/plugins/shared/stack_connectors/server/connector_types/slack_api/service.ts

```diff
@@ -165,46 +190,60 @@ export const createExternalService = (
     }
   };
 
+  const validateChannels = (channels?: string[], allowedList?: string[]) => {
+    if (!channels?.length || !allowedList?.length) return;
+
+    const hasDisallowedChannel = channels?.some((name) => !allowedList.includes(name));
+
+    if (hasDisallowedChannel) {
+      throw new Error(
+        `One or more provided channel names are not included in the allowed channels list`
+      );
+    }
+  };
+
   const getChannelToUse = ({
     channels,
     channelIds = [],
+    channelNames = [],
   }: {
     channels?: string[];
     channelIds?: string[];
+    channelNames?: string[];
   }): string => {
-    if (
-      channelIds.length > 0 &&
-      allowedChannelIds &&
-      allowedChannelIds.length > 0 &&
-      !channelIds.every((cId) => allowedChannelIds.includes(cId))
-    ) {
-      throw new Error(
-        `One of channel ids "${channelIds.join()}" is not included in the allowed channels list "${allowedChannelIds.join()}"`
-      );
+    const hasChannels =
+      channelNames.length > 0 || channelIds.length > 0 || (channels?.length ?? 0) > 0;
+
+    if (!hasChannels) {
+      throw new Error(`The channel is empty`);
+    }
+    // priority: channelNames > channelIds
+    if (channelNames.length > 0) {
+      validateChannels(channelNames, allowedChannelNames);
+      return channelNames[0];
     }
 
-    // For now, we only allow one channel but we wanted
-    // to have a array in case we need to allow multiple channels
-    // in one actions
-    let channelToUse = channelIds.length > 0 ? channelIds[0] : '';
-    if (channelToUse.length === 0 && channels && channels.length > 0 && channels[0].length > 0) {
-      channelToUse = channels[0];
+    if (channelIds.length > 0) {
+      validateChannels(channelIds, allowedChannelIds);
+      return channelIds[0];
     }
 
-    if (channelToUse.length === 0) {
-      throw new Error(`The channel is empty"`);
+    if (channels && channels.length > 0) {
+      return channels[0];
     }
 
-    return channelToUse;
+    throw new Error(`The channel is empty`);
   };
 
   const postMessage = async ({
     channels,
     channelIds = [],
+    channelNames = [],
     text,
   }: PostMessageSubActionParams): Promise<ConnectorTypeExecutorResult<unknown>> => {
     try {
-      const channelToUse = getChannelToUse({ channels, channelIds });
+      validateChannelNames(channelNames);
```

Comment:
I think we could probably move all calls to `validateChannelNames` one level up.

https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/stack_connectors/server/connector_types/slack_api/index.ts#L103

The slack api executor does 3 possible calls. We put this validation in each of them. If the validation is moved up, it only gets called once.